go/tools/microbench/microbench.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package microbench
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/vitessio/arewefastyet/go/mysql"
+	"go/types"
+	"golang.org/x/tools/go/packages"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const (
+	errorInvalidProfileType = "invalid profile type"
+
+	profileCPU = "cpu"
+	profileMem = "mem"
+)
+
+type benchmark struct {
+	id               int64
+	filePath         string
+	name             string
+	pkgPath, pkgName string
+	sql              *mysql.Client
+}
+
+func (b *benchmark) registerToMySQL(client *mysql.Client) error {
+	query := "INSERT INTO microbenchmark(test_no, pkg_name, name) VALUES(?, ?, ?)"
+	id, err := client.Insert(query, 0, b.pkgName, b.name)
+	if err != nil {
+		return err
+	}
+	b.id = id
+	return nil
+}
+
+func (b *benchmark) execute(rootDir string, w *os.File) error {
+	command := exec.Command("go", "test", "-bench=^"+b.name+"$", "-run==", "-json", "-count=10", b.pkgPath)
+	command.Dir = rootDir
+	out, err := command.Output()
+
+	if err != nil {
+		return err
+	}
+
+	if b.sql != nil {
+		if err := b.registerToMySQL(b.sql); err != nil {
+			log.Fatal(err)
```

Comment:
nit: shouldn't you be returning the error here?

Reply:
I should! Fixed in 710b360